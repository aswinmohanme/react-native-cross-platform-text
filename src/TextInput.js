import React from 'react';
import { TextInput as ReactNativeTextInput } from 'react-native';
import FontStyleWrapper from './FontStyleWrapper';

const TextInput = ({ style, ...props }) => (
  <FontStyleWrapper style={style}>
    {fontStyle => <ReactNativeTextInput style={fontStyle} {...props} />}
  </FontStyleWrapper>
);

export default TextInput;
