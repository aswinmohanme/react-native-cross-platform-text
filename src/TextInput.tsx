import React from "react";
import { StyleProp, TextInput as ReactNativeTextInput, TextStyle } from "react-native";
import FontStyleWrapper from "./FontStyleWrapper";

interface TextInputProps {
  style: StyleProp<TextStyle>;
}

const TextInput = ({ style, ...props }: TextInputProps) => (
  <FontStyleWrapper style={style}>
    {(fontStyle) => <ReactNativeTextInput style={fontStyle} {...props} />}
  </FontStyleWrapper>
);

export default TextInput;
