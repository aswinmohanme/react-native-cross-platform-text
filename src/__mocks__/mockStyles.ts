import { StyleSheet } from 'react-native';

const doMockStyles = StyleSheet.create({
  fontStyle: {
    fontWeight: 'normal',
    fontFamily: 'Arial',
  },
  section: {
    paddingHorizontal: 10,
  },
  container: {
    margin: 20,
  },
});

export default doMockStyles;
