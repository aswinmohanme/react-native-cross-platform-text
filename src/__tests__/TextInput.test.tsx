import * as React from "react";
import { Platform } from "react-native";
import * as renderer from 'react-test-renderer';
import TextInput from "../TextInput";
import mockStyles from '../__mocks__/mockStyles';

jest.mock('Platform');

it("iOS returns fontFamily and weight for normal", () => {
  Platform.OS = 'ios';
  const tree = renderer.create(<TextInput style={mockStyles.fontStyle}>this is iOS</TextInput>);
  expect(tree).toMatchSnapshot();
});

it("Android returns fontFamily-Regular and weight = undefined for normal", () => {
  Platform.OS = 'android';
  const tree = renderer.create(<TextInput style={mockStyles.fontStyle}>this is Android</TextInput>);
  expect(tree);
});

it("flattens styles", () => {
  const tree = renderer.create(
    <TextInput
      style={[
        mockStyles.section,
        [mockStyles.container, mockStyles.fontStyle],
      ]}
    >works with nested styles!</TextInput>
  );
  expect(tree).toMatchSnapshot();
});

it("passes TextInput props through", () => {
  const tree = renderer.create(
    <TextInput
      style={mockStyles.fontStyle}
      defaultValue="Amazing"
    >passes props through!</TextInput>
  );
  expect(tree).toMatchSnapshot();
});
