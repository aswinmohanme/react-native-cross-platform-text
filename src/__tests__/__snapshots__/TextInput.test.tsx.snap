// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Android returns fontFamily-Regular and weight = undefined for normal 1`] = `
<TextInput
  allowFontScaling={true}
  style={
    Object {
      "fontFamily": "Arial-Regular",
      "fontWeight": undefined,
    }
  }
>
  this is Android
</TextInput>
`;

exports[`flattens styles 1`] = `
<TextInput
  allowFontScaling={true}
  style={
    Object {
      "fontFamily": "Arial",
      "fontWeight": "normal",
      "margin": 20,
      "paddingHorizontal": 10,
    }
  }
>
  works with nested styles!
</TextInput>
`;

exports[`iOS returns fontFamily and weight for normal 1`] = `
<TextInput
  allowFontScaling={true}
  style={
    Object {
      "fontFamily": "Arial",
      "fontWeight": "normal",
    }
  }
>
  this is iOS
</TextInput>
`;

exports[`passes TextInput props through 1`] = `
<TextInput
  allowFontScaling={true}
  defaultValue="Amazing"
  style={
    Object {
      "fontFamily": "Arial",
      "fontWeight": "normal",
    }
  }
>
  passes props through!
</TextInput>
`;
